fall back to zram if no swap partition or file was found

dimkr opened the pull request.

dimkr committed:
fall back to 1GB of zram if no swap partition or file was found

dimkr committed:
choose zram size the same way Chrome OS does on ARM

01micko commented on woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:
```diff
@@ -60,8 +60,8 @@ status_func() {
   modprobe zram 2>/dev/null
-  # compression ratio should be in the range of 2:1 to 3:1; 1 GB will bring 32-bit machines with 2 GB of RAM to the limit of 4 GB
-  echo 1G > /sys/block/zram0/disksize || return 1
+  RAM=`awk '/MemTotal/{print $2}' /proc/meminfo`
```
Better idea!

01micko commented on woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:
```diff
@@ -58,6 +58,14 @@ status_func() {
  return $1 #return with same status as entry param.
 }
 
+create_zram_swap() {
```
perhaps we `return 1` if we run an older kernel not supporting zram here?

dimkr replied:
zram can be builtin. The `echo` will fail, if zram wasn't loaded and isn't builtin.

01micko commented on woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:
```diff
@@ -58,6 +58,14 @@ status_func() {
 }
 
+create_zram_swap() {
+  modprobe zram 2>/dev/null
```
woops, should be this line
`modprobe zram 2>/dev/null || return 1`

dimkr replied:
See my previous comment

01micko replied:
Ok, I'll remove the `|| return 1` from 2066bed

dimkr commented:
@01micko Was it your intention to close this PR without merging it? Personally, I want it merged (great for my 2 GB laptop), but I'm open to feedback and even objections.

01micko commented:
Not at all! I changed it and added credit to you in 2066bed Then you indicated that maybe zram could be builtin so I fixed it, so my bad. If you want, I can revert my commits and just pull the PR!

dimkr committed:
Merge branch 'testing' of https://github.com/puppylinux-woof-CE/woof-CE into feature/zram-if-no-swap-found

dimkr commented:
> Not at all! I changed it and added credit to you in [2066bed](https://github.com/puppylinux-woof-CE/woof-CE/commit/2066bed3081f683b76272b51fa5cfbac25dd0b76) Then you indicated that maybe zram could be builtin so I fixed it, so my bad. If you want, I can revert my commits and just pull the PR!

Thanks, I got confused. I thought you pushed a commit to this PR, and not to testing, then closed the PR.
